src/stdlib_ascii.f90

```diff
@@ -145,13 +145,15 @@ pure logical function is_printable(c)
         character(len=1), intent(in) :: c !! The character to test.
         integer :: ic
         ic = iachar(c)                    ! '~'
```

Comment:
`z'7E' = '~'`
I suggest either adding some spaces so the character is aligned again 
(or removing it and adding a explicit comment in a new line showing how we are checking it)

```suggestion
        ic = iachar(c)
        !The character is printable if it's between ' ' and '~' in the ASCII table
```
What do you think?